plugins/module_utils/ldap.py

```diff
@@ -58,13 +58,32 @@ def __init__(self, module):
         # Establish connection
         self.connection = self._connect_to_ldap()
 
+        # Try to find the X_ORDERed version of the DN
+        self.dn = self._find_dn()
+
     def fail(self, msg, exn):
         self.module.fail_json(
             msg=msg,
             details=to_native(exn),
             exception=traceback.format_exc()
         )
 
+    def _find_dn(self):
+        dn = self.module.params['dn']
+
+        explode_dn = ldap.dn.explode_dn(dn)
+
+        if len(explode_dn) > 1:
+            try:
+                dns = self.connection.search_s(','.join(explode_dn[1:]),
+                                               ldap.SCOPE_ONELEVEL, "(%s)" % explode_dn[0])
```

Comment:
Maybe consider escaping the filterstr using `ldap.filter.escape_filter_chars` (same as in https://github.com/ansible-collections/community.general/pull/5435). Although the module wouldn't fail thanks to the exception handling below, it is probably still a good idea to escape the filter and thus make sure that the `{x}` prefix handling works correct even if there is a special character in the DN.